motor/core.py

```diff
@@ -1042,22 +1048,13 @@ def main():
                       change_stream.close()
 
           # asyncio
-          from asyncio import get_event_loop
-
-          def main():
-              loop = get_event_loop()
-              task = loop.create_task(watch_collection)
-
-              try:
-                  loop.run_forever()
-              except KeyboardInterrupt:
+          try:
+              asyncio.run(watch_collection)
+          except KeyboardInterrupt:
                   pass
-              finally:
-                  if change_stream is not None:
-                      change_stream.close()
-
-                  # Prevent "Task was destroyed but it is pending!"
-                  loop.run_until_complete(task)
+          finally:
+              if change_stream is not None:
+                  change_stream.close()
```

Comment:
change_stream.close() is an async method so calling it here leaks a coroutine. It's fine for this PR since it's a preexisting issue but could you file a new issue to improve this code example? Really it should work via task cancellation, not through KeyboardInterrupt.

Reply:
I opened https://jira.mongodb.org/browse/MOTOR-938

Reply:
Thanks, sorry I meant to return to this comment and missed it in the scroll